pkg/credentialprovider/keyring.go

```diff
@@ -57,11 +57,11 @@ type lazyDockerKeyring struct {
 func (dk *BasicDockerKeyring) Add(cfg DockerConfig) {
 	if dk.index == nil {
 		dk.index = make([]string, 0)
-		dk.creds = make(map[string][]docker.AuthConfiguration)
+		dk.creds = make(map[string][]dockertypes.AuthConfig)
 	}
 	for loc, ident := range cfg {
 
-		creds := docker.AuthConfiguration{
+		creds := dockertypes.AuthConfig{
```

Comment:
We should not directly expose an unversioned type from a client library we
import to our users, because then changes to that library become breaking
changes to our users.

On Wed, Apr 13, 2016 at 10:57 AM, David Eads <notifications@github.com>
wrote:

> In pkg/credentialprovider/keyring.go
> <https://github.com/kubernetes/kubernetes/pull/23800#discussion_r59594703>
> :
>
> >  	}
> >  	for loc, ident := range cfg {
> >
> > -		creds := docker.AuthConfiguration{
> > +		creds := dockertypes.AuthConfig{
>
> @deads2k <https://github.com/deads2k> I have only very foggy recollection
> of this code. Was this a FYI or are you asking me for something?
>
> They're talking about removing the kube serialized type as some kind of
> followup. I'd like to keep our specific serialized type, but figured you'd
> have an opinion.
>
> @lavalamp <https://github.com/lavalamp> Should we keep our serialization
> type to control the format and simply write tests to track drift between
> our types and upstream types? (we do this in openshift and it works
> reasonably well)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23800/files/4c318ea8f212fa1186660e369c61ad087e1d7666#r59594703>
>